include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,204 @@
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsT(n_workspaces, dist_t);
+
+  // Specific for V matrix layout where last column of tiles is empty
+  const SizeType num_panel_refls = mat_v.nrTiles().cols() - 1;
+  const SizeType last_tile_cols = mat_v.tileSize(GlobalTileIndex(0, m - 1)).cols();
+
+  for (SizeType k = num_panel_refls - 1; k >= 0; --k) {
+    bool is_last = (k == num_panel_refls - 1) ? true : false;
+    const SizeType t_last_cols = dist_t.tileSize(GlobalTileIndex{k, k}).cols();
+
+    auto& panelV = panelsV.nextResource();
+    auto& panelW = panelsW.nextResource();
+    auto& panelW2 = panelsW2.nextResource();
+    auto& panelT = panelsT.nextResource();
+
+    panelV.setRangeStart({k + 1, k});
+    panelW.setRangeStart({k + 1, k});
+
+    for (SizeType i = k + 1; i < mat_v.nrTiles().rows(); ++i) {
+      // Copy V panel into VV
+      auto ik = LocalTileIndex{i, k};
+
+      // Setting VV
+      if (i == k + 1) {
+        copySingleTile(mat_v.read(ik), panelV(ik));
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::laset<T>), lapack::MatrixType::Upper, 0.f, 1.f,
+                      panelV(ik));
+      }
+      else {
+        hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_v = mat_v.read(ik);
+
+        if (is_last) {
+          tile_v = splitTile(tile_v, {{0, 0}, dist_t.tileSize(GlobalTileIndex(k, k))});
+        }
+
+        panelV.setTile(ik, mat_v.read(ik));
+      }
+
+      // Copy VV into W
+      copySingleTile(panelV.read(ik), panelW(ik));
+    }
+
+    // Set panelW2 to zero
+    matrix::util::set0(executor_hp, panelW2);
+
+    const GlobalTileIndex v_start{k + 1, k};
+    auto taus_panel = taus[k];
+    const SizeType taupan = (is_last) ? last_tile_cols : mat_v.blockSize().cols();
+    auto kk = LocalTileIndex{k, k};
+    const LocalTileIndex diag_wp_idx{Coord::Col, k};
+
+    if (is_last) {
+      panelT.setHeight(t_last_cols);
+      panelW2.setHeight(t_last_cols);
+      panelW.setWidth(last_tile_cols);
+      panelV.setWidth(last_tile_cols);
+    }
+
+    dlaf::factorization::internal::computeTFactor<Backend::MC>(taupan, mat_v, v_start, taus_panel,
+                                                               panelT(diag_wp_idx));
+
+    // WH = V T
+    hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_t = panelT.read(diag_wp_idx);
+    for (const auto& idx : panelW.iteratorLocal()) {
+      trmmPanel(executor_np, tile_t, panelW(idx));
+    }
+
+    // W2 = W C
+    for (SizeType j = 0; j < n; ++j) {
+      auto kj = LocalTileIndex{k, j};
+      for (SizeType i = k + 1; i < m; ++i) {
+        auto ik = LocalTileIndex{i, k};
+        auto ij = LocalTileIndex{i, j};
+        gemmUpdateW2(executor_np, panelW(ik), mat_c.read(ij), panelW2(kj));
+      }
+    }
+
+    // C = C - V W2
+    LocalTileIndex c_start{k + 1, 0};
+    LocalTileIndex c_end{m, n};
+    for (const auto& idx : iterate_range2d(c_start, c_end)) {

```

Comment:
I think it would be way more clear if you define what this represents (and at the beginning at the algorithm together with other indices relevant for the algorithm).

It looks like you are defining the matrix `C_k`, i.e. the C submatrix you are going to work whose shape depends on the iteration of the algorithm. Indeed, you are defining it in terms of `k+1` (see my previous comment) and you could use it also for the previous calculation of `W2`.

Moreover, once you will define this range describing the C_k submatrix, you could use the same index you use for `mat_c` also for the panels (as already explained in previous reviews).